src/rust/ide/json-rpc/src/macros.rs

```diff
@@ -0,0 +1,158 @@
+//! Helper macros to generate RemoteClient and MockClient.
+
+
+
+// ====================
+// === Helper macro ===
+// ====================
+
+/// Generates parameters maps. e.g.:
+/// `(age:u32)` => `u32`
+/// `(age:u32,height:f32)` => `(u32,f32)`
+#[macro_export]
+macro_rules! make_param_map {
+    (,$ty:ty) => {
+        $ty
+    };
+    (,$($ty:ty),+) => {
+        ($($ty),+)
+    }
+}
+
+/// Generates arguments. e.g.:
+/// `(age:u32)` => `age`
+/// `(age:u32,height:f32)` => `(age,height)`
+#[macro_export]
+macro_rules! make_arg {
+    ($name:ident) => {
+        $name
+    };
+    ($($name:ident),+) => {
+        ($($name),+)
+    }
+}
```

Comment:
Again, the docs are not true here. This macro does not what docs tell and its a big hack instead.